oio/api/backblaze_http.py

```diff
@@ -37,6 +38,47 @@ def _get_sha1(data):
     return generate.hexdigest()
 
 
+class BackblazeUtils(object):
+
+    b2_authorization_list = {}
+
+    @staticmethod
+    def put_meta_backblaze(storage_method, application_key_path=None,
+                           renew=False):
+        if not (storage_method.bucket_name and
+                storage_method.account_id and application_key_path):
+            message = "%s (bucket_name=%s, account_id=%s)" % \
+                      ('missing backblaze parameters: ',
+                       storage_method.bucket_name, storage_method.account_id)
+            raise BackblazeUtilsException(message)
+        key = '%s.%s' % (storage_method.account_id, storage_method.bucket_name)
+        if not renew:
+            authorization = BackblazeUtils.b2_authorization_list.get(key, None)
+            if authorization:
+                return authorization
+        config = ConfigParser.ConfigParser()
+        try:
+            config.readfp(open(application_key_path))
```

Comment:
The file is never closed. You should use a _with_ statement or close is in a _finally_ block.